RecoLocalMuon/MuonSysAging/plugins/GEMChamberMasker.cc

```diff
@@ -0,0 +1,172 @@
+#include "DataFormats/GEMDigi/interface/GEMPadDigiCollection.h"
+#include "DataFormats/GEMDigi/interface/GEMCoPadDigiCollection.h"
+#include "DataFormats/MuonDetId/interface/GEMDetId.h"
+#include "CondFormats/RecoMuonObjects/interface/MuonSystemAging.h"
+#include "CondFormats/DataRecord/interface/MuonSystemAgingRcd.h"
+
+//
+// class declaration
+//
+
+class GEMChamberMasker : public edm::stream::EDProducer<>
+{
+
+   public:
+      explicit GEMChamberMasker(const edm::ParameterSet&);
+      ~GEMChamberMasker();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      virtual void produce(edm::Event&, const edm::EventSetup&) override;
+      
+      virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+      virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+
+      // ----------member data ---------------------------
+  edm::InputTag digiTag_;
+  bool GE11Minus_;
+  bool GE11Plus_; 
+  bool GE21Minus_;
+  bool GE21Plus_; 
+
+  edm::EDGetTokenT<GEMDigiCollection> m_digiTag;
+  std::map<unsigned int, float> m_maskedGEMIDs;
+
+};
+
+//
+// constants, enums and typedefs
+//
+
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+GEMChamberMasker::GEMChamberMasker(const edm::ParameterSet& iConfig) : 
+  digiTag_(iConfig.getParameter<edm::InputTag>("digiTag") ),
+  GE11Minus_(iConfig.getParameter<bool>("GE11Minus") ),
+  GE11Plus_(iConfig.getParameter<bool>("GE11Plus") ),
+  GE21Minus_(iConfig.getParameter<bool>("GE21Minus") ),
+  GE21Plus_(iConfig.getParameter<bool>("GE21Plus") )
+{
+
+  m_digiTag = consumes<GEMDigiCollection>(digiTag_); 
+  produces<GEMDigiCollection>();
+  
+}
+
+
+GEMChamberMasker::~GEMChamberMasker()
+{ 
+
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void
+GEMChamberMasker::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+
+  using namespace edm;
+  std::unique_ptr<GEMDigiCollection> filteredDigis(new GEMDigiCollection());
+
+  if (!digiTag_.label().empty())
+    {
+      edm::Handle<GEMDigiCollection> gemDigis;
+      iEvent.getByToken(m_digiTag, gemDigis);
+      
+      GEMDigiCollection::DigiRangeIterator gemLayerIdIt  = gemDigis->begin();
+      GEMDigiCollection::DigiRangeIterator gemLayerIdEnd = gemDigis->end();
+
+
+      
+      for (; gemLayerIdIt != gemLayerIdEnd; ++gemLayerIdIt)
+	{
+	  auto chambId = (*gemLayerIdIt).first.chamberId();
+
+	  bool keepDigi = (!GE11Minus_  && chambId.station()==1 && chambId.region()<0 ) ||
+		          (!GE11Plus_   && chambId.station()==1 && chambId.region()>0 ) ||
+		          (!GE21Minus_  && chambId.station()==2 && chambId.region()<0 ) ||
+		          (!GE21Plus_   && chambId.station()==2 && chambId.region()>0 ) ;
+
+          uint32_t rawId = chambId.rawId();
+          if(keepDigi || m_maskedGEMIDs.find(rawId) == m_maskedGEMIDs.end())
```

Comment:
so, m_GEMChambEffs is meant only for on and off?
What's the logic to keep it different from DT and CSC functionality?